### PR TITLE
Added support for Facets in the Single Search page response

### PR DESCRIPTION
**Pull request by jpuri08**

### CODE REVIEW

**Comment by dnfgituser** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1082) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1082) before we can accept your contribution.<br/><br/>:x: jpuri08 [sign now](https://cla.dotnetfoundation.org/NuGet/NuGet.Jobs?pullRequest=1082)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/NuGet/NuGet.Jobs?pullRequest=1082) it.</sub>

File: src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Net;
-using System.Threading.Tasks;
 using Azure;
 using Azure.Search.Documents;
 using Azure.Search.Documents.Models;
 using NuGet.Services.AzureSearch.SearchService;
+using System;
```
Comment (zhhyu) [inline src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs:8]: nit: please always put "System.*" libraries as the first ones.

**Review by shishirx34 — DISMISSED** [review]
@agr for a second look.

**Review by agr — APPROVED** [review]
LGTM, but please reorder the usings and sign CLA (link in the first comment).

File: src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs (hunk 1 above)
Comment (jpuri08) [reply src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs:4]: Done.

File: src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs [hunk 2]
```diff
@@ -49,15 +50,15 @@ public SearchClientWrapper(SearchClient inner)
                         while (await enumerator.MoveNextAsync())
                         {
                             // Only read the first page. We use paging parameters through the application to avoid paging.
-                            return new SingleSearchResultPage<T>(enumerator.Current.Values, results.TotalCount);
+                            return new SingleSearchResultPage<T>(enumerator.Current.Values, results.TotalCount, results.Facets);
```
[the remark this answers is not in the document]
Comment (jpuri08) [reply src/NuGet.Services.AzureSearch/Wrappers/SearchClientWrapper.cs:53]: The constructor call will automatically be covered in unit test cases. NuGet search is not using Facet, so we don't need any specific unit test cases related to change. The VS Code search service is using Facets and test cases are already covered.

**Review by zhhyu — APPROVED** [review]

File: src/NuGet.Services.AzureSearch/Wrappers/SingleSearchResultPage.cs [hunk 3]
```diff
@@ -14,7 +14,16 @@ public SingleSearchResultPage(IReadOnlyList<SearchResult<T>> values, long? count
...
+        public SingleSearchResultPage(IReadOnlyList<SearchResult<T>> values, long? count, IDictionary<string, IList<FacetResult>> facets)
+        {
+            Values = values;
+            Count = count;
+            Facets = facets;
+        }
+
         public IReadOnlyList<SearchResult<T>> Values { get; set; }
         public long? Count { get; }
+        public IDictionary<string, IList<FacetResult>> Facets { get; set; }
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureSearch/Wrappers/SingleSearchResultPage.cs:26]: Should this use read only collections?

```suggestion
        public IReadOnlyDictionary<string, IReadOnlyList<FacetResult>> Facets { get; set; }
```

**Comment by shishirx34** [comment]
@loic-sharma - Thank you for the review. Do you see any issues with merging this PR, any caveats for search service?

**Comment by loic-sharma** [comment]
This change looks good to me

File: src/NuGet.Services.AzureSearch/Wrappers/SingleSearchResultPage.cs (hunk 3 above)
Comment (jpuri08) [reply src/NuGet.Services.AzureSearch/Wrappers/SingleSearchResultPage.cs:26]: The Datatype for facets in response from SDK client is **IDictionary<string, IList<FacetResult>> Facets** . I need to manually convert all the lists into the read-only list for each entry in the dictionary while creating the SingleSearchResultPage object in the searchClientWrapper.
ScreenShot_17-08-2022_18_42_26

**Pull request merged**